migrate/repo_checks.py

```diff
@@ -54,6 +75,93 @@ def dry_run(self):
         raise NotImplementedError
 
 
+class RequireTeamPermission(Check):
+    """
+    A generic check that is not meant to be used directly
+    but is used to reduce some copy-pasta.
+
+    Do not use directly, instead, subclass this class with
+    a specific team and permission level.
+    """
```

Comment:
I think extracting this class was a good refactoring, but the docstring confuses me. It seems like it'd be fine to initialize and use this class directly; in fact you do that below in `RequireCLACheck`.